concordium-sdk/src/main/java/com/concordium/sdk/Client.java

```diff
@@ -325,6 +326,48 @@ public Semver getVersion() {
         return new Semver(versionString);
     }
 
+    /**
+     * Ban a specific node.
+     * Note that this will also cause the node to drop any connections to a matching node.
+     *
+     * @param request {@link BanNodeRequest}
+     * @return {@link Boolean} depending on whether the request was successful or not.
+     * @throws Exception When Neither ID nor Ip was specified.
+     */
+    public boolean banNode(final BanNodeRequest request) throws Exception {
+        ConcordiumP2PRpc.PeerElement peerElement;
+
+        if (request.hasIp()) {
+            peerElement = ConcordiumP2PRpc.PeerElement.newBuilder()
+                    .setIp(StringValue.of(request.getIp().getHostAddress()))
+                    .build();
+
+        } else if (request.hasId()) {
+            peerElement = ConcordiumP2PRpc.PeerElement
+                    .newBuilder()
+                    .setNodeId(StringValue.of(request.getId()))
+                    .build();
+        } else {
+            throw new Exception("Invalid Request. Should mention either Ip or Id");
```

Comment:
I don't think this `Exception` is necessary as you are already enforcing that either the IP or the id is set in the request.

Reply:
removed